### PR TITLE
Create componentconfig object for scheduler configuration.

### PR DESCRIPTION
**Pull request by mikedanese**
This is part of moving scheduler to componentconfig

### CODE REVIEW

**Comment by mikedanese** [comment]
@davidopp Thanks for reviewing all my PRs! Feel free to reassign

File: docs/admin/kube-scheduler.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ kube-scheduler
       --kube-api-qps=50: QPS to use while talking with kubernetes apiserver
       --kubeconfig="": Path to kubeconfig file with authorization and master location information.
       --leader-elect[=false]: Start a leader election client and gain leadership before executing scheduler loop. Enable this when running replicated schedulers.
-      --leader-elect-lease-duration=15s: The duration that non-leader candidates will wait after observing a leadership renewal until attempting to acquire leadership of a led but unrenewed leader slot. This is effectively the maximum duration that a leader can be stopped before it is replaced by another candidate. This is only applicable if leader election is enabled.
+      --leader-elect-lease-duration=15s: The duration that non-leader candidates will wait after observing a leadershiprenewal until attempting to acquire leadership of a led but unrenewed leader slot. This is effectively the maximum duration that a leader can be stopped before it is replaced by another candidate. This is only applicable if leader election is enabled.
```
Comment (davidopp) [inline docs/admin/kube-scheduler.md]: is there a reason why you made "leadershiprenewal' all one word?

Comment (mikedanese) [inline docs/admin/kube-scheduler.md]: Fixed.

**Comment by mikedanese** [comment]
@davidopp fixed

File: plugin/cmd/kube-scheduler/app/options/options.go [hunk 2]
```diff
@@ -63,15 +57,17 @@ func NewSchedulerServer() *SchedulerServer {
...
-	fs.IPVar(&s.Address, "address", s.Address, "The IP address to serve on (set to 0.0.0.0 for all interfaces)")
+	fs.StringVar(&s.Address, "address", s.Address, "The IP address to serve on (set to 0.0.0.0 for all interfaces)")
 	fs.StringVar(&s.AlgorithmProvider, "algorithm-provider", s.AlgorithmProvider, "The scheduling algorithm provider to use, one of: "+factory.ListAlgorithmProviders())
 	fs.StringVar(&s.PolicyConfigFile, "policy-config-file", s.PolicyConfigFile, "File with scheduler policy configuration")
 	fs.BoolVar(&s.EnableProfiling, "profiling", true, "Enable profiling via web interface host:port/debug/pprof/")
 	fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
 	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
-	fs.Float32Var(&s.BindPodsQPS, "bind-pods-qps", s.BindPodsQPS, "Number of bindings per second scheduler is allowed to continuously make")
+	var unusedBindPodsQPS float32
+	fs.Float32Var(&unusedBindPodsQPS, "bind-pods-qps", 0, "unused, use --pind-pods-qps")
```
Comment (davidopp) [inline plugin/cmd/kube-scheduler/app/options/options.go]: s/--pind-pods-qps/--bind-pods-qps/

File: pkg/apis/componentconfig/types.go [hunk 3]
```diff
@@ -278,6 +278,30 @@ type KubeletConfiguration struct {
...
+	EnableProfiling bool `json:"enableProfiling"`
+	// kubeAPIQPS is the QPS to use while talking with kubernetes apiserver.
+	KubeAPIQPS float32 `json:"kubeAPIQPS"`
+	// kubeAPIBurst is the QPS burst to use while talking with kubernetes apiserver.
+	KubeAPIBurst int `json:"kubeAPIBurst"`
+	// schedulerName is name of the scheduler, used to select which pods
+	// will be processed by this scheduler, based on pod's annotation with
+	// key 'scheduler.alpha.kubernetes.io/name'.
+	SchedulerName string `json:"schedulerName"`
+	// leaderElection defines the configuration of leader election client
```
Comment (davidopp) [inline pkg/apis/componentconfig/types.go]: one of the two "configuration" should be removed

File: plugin/cmd/kube-scheduler/app/options/options.go [hunk 4]
```diff
@@ -63,15 +57,17 @@ func NewSchedulerServer() *SchedulerServer {
...
-	fs.IPVar(&s.Address, "address", s.Address, "The IP address to serve on (set to 0.0.0.0 for all interfaces)")
+	fs.StringVar(&s.Address, "address", s.Address, "The IP address to serve on (set to 0.0.0.0 for all interfaces)")
 	fs.StringVar(&s.AlgorithmProvider, "algorithm-provider", s.AlgorithmProvider, "The scheduling algorithm provider to use, one of: "+factory.ListAlgorithmProviders())
 	fs.StringVar(&s.PolicyConfigFile, "policy-config-file", s.PolicyConfigFile, "File with scheduler policy configuration")
 	fs.BoolVar(&s.EnableProfiling, "profiling", true, "Enable profiling via web interface host:port/debug/pprof/")
 	fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
 	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
-	fs.Float32Var(&s.BindPodsQPS, "bind-pods-qps", s.BindPodsQPS, "Number of bindings per second scheduler is allowed to continuously make")
+	var unusedBindPodsQPS float32
+	fs.Float32Var(&unusedBindPodsQPS, "bind-pods-qps", 0, "unused, use --bind-pods-qps")
```
Comment (davidopp) [inline plugin/cmd/kube-scheduler/app/options/options.go]: pretty sure the description string is wrong

**Comment by davidopp** [comment]
LGTM

**Comment by davidopp** [comment]
@k8s-bot test this please

**Comment by mikedanese** [comment]
@k8s-bot e2e test this please

**Comment by mikedanese** [comment]
@k8s-bot e2e test this please

**Pull request merged**